boards/arm/twr_ke18f/twr_ke18f.dts

```diff
@@ -15,6 +15,17 @@
 	model = "NXP Kinetis KE18 MCU Tower System Module";
 	compatible = "nxp,mke18f16", "nxp,ke18f", "nxp,ke1xf";
 
+	/* Additional pinctrl states for flexio, as well as SPI PCS pins */
+	zephyr,user {
+		/* FlexIO clockout pin */
+		flexio_clockout_mux = <&flexio_clockout>;
+		/* LPSPI PCS muxes */
+		lpspi0_pcs2_mux = <&lpspi0_pcs2>;
+		lpspi1_pcs0_mux = <&lpspi1_pcs0>;
+		lpspi1_pcs2_mux = <&lpspi1_pcs2>;
+		lpspi1_pcs0_pcs2_mux = <&lpspi1_pcs0_pcs2>;
+	};
```

Comment:
note: I'm not familiar with this board, but maybe you can provide overlays that select different choices instead of using dynamic pinctrl (or maybe board variants?). Dynamic pinctrl is used when the selection is not known beforehand, e.g. depends on input like a GPIO state.

Reply:
I think overlays would be the best option here. I know @henrikbrixandersen is relatively familiar with this board. @henrikbrixandersen, I'll update this PR to use overlays, and if that does not seem like an acceptable solution to you let me know and we can adjust.

Reply:
For the FlexIO clock pin, what would be the appropriate way to apply the pinmux? There is no upstream driver for NXP's FlexIO peripheral, but it does not fit the use case of dynamic pinctrl.

Reply:
Overlays sounds good, but please do not spend too much time on it. I can always fix it up later or use local overlays as needed. I doubt anybody else are using the board specific pinctrl Kconfig options.

Reply:
After playing with overlays for a bit, I think we might need to stick to dynamic pinctrl. The search directory list for devicetree overlays doesn't include the board directory, and even if it did the board `CMakeLists.txt` is parsed too late in the build process to add a devicetree overlay.

Reply:
Btw, an example of a board that provides some extra overlays: https://github.com/zephyrproject-rtos/zephyr/tree/main/boards/arm/particle_xenon/dts

Reply:
Thanks for the example, I've got overlays working by moving them to a `dts` folder in the board directory. The flexIO pin still is enabled using dynamic pinctrl. Generally speaking, is there a way to apply pins outside of a driver definition without using dynamic pin control?

Reply:
You can do one trick to achieve that (haven't tried, but I think it should work):

1. Create a board level binding at `${BOARD}/dts/bindings/` (see e.g. `nrf9160dk_nrf52840`), if it's not board specific maybe just put it under `dts/bindings`.
2. Define your flexIO entry in DT like this:
```
flexio: flexio {
    compatible = "....";
    pinctrl-0 = <&flexio_default>; /* add pin config in the `-pinctrl.dtsi` file */
    pinctrl-names = "default";
};
```
3. In your board.c you can then do:
```c
#define FLEXIO DT_NODELABEL(flexio)
PINCTRL_DT_DEFINE(FLEXIO);
ret = pinctrl_apply_state(PINCTRL_DT_DEV_CONFIG_GET(FLEXIO), PINCTRL_STATE_DEFAULT);
...
```